bosh-stemcell/spec/stemcells/fips_spec.rb

```diff
@@ -37,7 +39,11 @@
     describe file('/boot/grub/grub.cfg') do
       it { should be_file }
       its(:content) { should match %r{linux\t/boot/vmlinuz-\S+-fips root=UUID=\S* ro } }
-      its(:content) { should match %r{initrd\t/boot/initrd.img-\S+-fips} }
+      if ENV.key?("UBUNTU_FIPS_USE_IAAS_KERNEL")
```

Comment:
Have you checked if the else can be reached? It could be that this environment variable always will be set, just to an empty string. Because of: https://github.com/cloudfoundry/bosh-linux-stemcell-builder/blob/04e622ab133a1d88a4a8b4b4685498925372c33e/bosh-stemcell/lib/bosh/stemcell/build_environment.rb#L54

Reply:
sorry I had to put this down for a couple weeks I've added a check for empty string as well as I think you're right about all the envvars being 'set' regardless